### ISSUE TITLE
Issue in first word in zipformer2

### ISSUE THREAD

**Issue by bhaswa** [k2-fsa/sherpa-onnx#348]
I found that in many cases the first word of a audio is not able to get decoded properly. But when I use the .pth model in icefall instead of .onnx model, the word gets decoded properly.

**Comment by csukuangfj** [comment]
Are you using the latest icefall to export the model and also are you using the latest sherpa-onnx for testing?

**Comment by bhaswa** [comment]
Any update on this ?

**Comment by bhaswa** [comment]
I observed this scenario in a model which I trained with custom data. The same audio might not behave the same way in another model (which is trained with different set of data).

**Comment by bhaswa** [comment]
Is there any pre trained model (.pth) available ? I will share a test wave file testing on that model.

**Comment by csukuangfj** [comment]
> Is there any pre trained model (.pth) available ? I will share a test wave file testing on that model.

Yes, please find the models in the RESULTS.md of each recipe in icefall, e.g., librispeech.
For each experiment, there is a link to the huggingface repo containing pre-trained models.

**Comment by bhaswa** [comment]
I am attaching two audio files here.

[audios.zip](https://github.com/k2-fsa/sherpa-onnx/files/12856262/audios.zip)

1.wav:
pth output(from icefall):                   I LIKE TO STOP RECEIVING NOTIFICATION FROM EASY TAXI
onnx output(from icefall):                I LIKE TO STOP RECEIVING NOTIFICATION FROM EASY TAXI
onnx output (from sherpa-onnx):   **HILA TWO STOP PRESLIGH NOTIFICATION FROM Y TAXI**
(pth and onnx output is matching in icefall but sherpa onnx output is different)

2.wav
pth output(from icefall):                   ALLOW PAST BOX TO SHOW NARTIFICATION BANNERS
onnx output(from icefall):                ALLOWS BOX TO SHOW NARTIFICATION BANNERS
onnx output (from sherpa-onnx):   **ILACAST BOXER TWO SHOIFICATION BANNERS**
(None of the outputs are matching here)

I used the model streaming zipformer (zipformer + pruned stateless transducer) [https://huggingface.co/Zengwei/icefall-asr-librispeech-streaming-zipformer-2023-05-17/tree/main/exp] from huggingface for testing these audios.

Command to convert the model to onnx is 

./zipformer/export-onnx-streaming.py \
  --exp-dir ./zipformer/exp \
  --tokens data/lang_bpe_500/tokens.txt \
  --causal 1 \
  --chunk-size 16 \
  --left-context-frames 128 \
  --epoch 30 \
  --avg 1 \
  --use-averaged-model False

**Comment by bhaswa** [comment]
Any update on this issue?

**Comment by kamirdin** [comment]
Zero key/value cache from encoder initialization states may be causing this discrepancy between training and decoding?
